src/App.js

```diff
@@ -4,37 +4,52 @@ import {
   writeNumber,
   readWinners,
   writeRacingCar,
-} from './domain/RacingIO';
+} from './view/RacingIO';
 import { readLineAsync } from './utils';
 
 async function App() {
-  try {
-    const racingCarNames = await readLineAsync(
-      '경주할 자동차 이름을 입력하세요(이름은 쉼표(,)를 기준으로 구분).\n'
-    );
-    const racingLapCount = await readLineAsync('시도할 회수는 몇회인가요?\n');
-
-    const racing = new Racing(writeNumber(racingLapCount));
-    racing.players = writeRacingCar(racingCarNames);
-
-    console.log('\n실행 결과');
-    while (!racing.isEndedRace()) {
-      racing.race1Lap();
-
-      for (const car of racing.players) {
-        console.log(readCarProgress(car));
-      }
-      console.log('\n');
+  let [isInValidNames, isInValidLapCount] = [true, true];
+  let racingCarNames = '';
+  let racing = new Racing();
+
+  while (isInValidNames) {
+    try {
+      racingCarNames = await readLineAsync(
+        '경주할 자동차 이름을 입력하세요(이름은 쉼표(,)를 기준으로 구분).\n'
+      );
+      racing.players = writeRacingCar(racingCarNames);
+
+      isInValidNames = false;
+    } catch (e) {
+      console.log(e.message + '\n');
+    }
+  }
+
+  while (isInValidLapCount) {
+    try {
+      const racingLapCount = await readLineAsync('시도할 회수는 몇회인가요?\n');
+      racing.maxLap = writeNumber(racingLapCount);
+
+      isInValidLapCount = false;
+    } catch (e) {
+      console.log(e.message + '\n');
     }
+  }
```

Comment:
사용자로부터 프롬프트 답변을 받는 로직이 절차적 프로그래밍되어 있어 가독성을 위해 모듈화가 필요해보여요. 

[enquirer](https://github.com/enquirer/enquirer?tab=readme-ov-file#prompt-options) 스타일로 프롬프트 로직을 작성할 수 있도록 구현해보시면 선언형 프로그래밍 연습이 되실 것 같습니다!

```suggestion
  const carNames = await prompt('경주할 자동차 이름을 입력하세요(이름은 쉼표(,)를 기준으로 구분).', {
    validate: validateCarnames,
    format: writeRacingCar,
  });
  const racingLapCount = await prompt('시도할 회수는 몇회인가요?', {
    validate: validateLapCount,
    format: writeNumber,
  });
```

Reply:
감사합니다 🙇  확실히 절차적 프로그래밍으로 되어있다보니 가독성이 떨어지네요!!
프롬프트 호출 과정중에 validate 검사 진행을 위하여, 추가적으로 에러 처리과정을 따로 분리하여 수정해보았습니다!